trackintel/similarity/detection.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+from trackintel.similarity import measures
+import numpy as np
+from scipy.sparse import dok_matrix
+from console_progressbar import ProgressBar
+
+
+def e_dist_tuples(c1,c2):
+    return np.sqrt((c1[0]-c2[0])**2 + (c1[1]-c2[1])**2)
+
+def similarity_matrix(data, method, field='tripleg_id', trsh=None, eps=None, dist=False, **kwargs):   
```

Comment:
I think there are some critical adjustments that have to be made before we can merge this.
- I don't think that tripleg similarity should be based on positionfixes with an ID column but only on the tripleg data (and then on the query points of the tripleg). This will allow us to cover use cases where we have triplegs but no positionfixes (e.g., GreenClass). 
- I don't think that the matrix assignment should be ID based. At the moment we are calculating a full distance matrix (meaning a distance between all pairs) but store it in a sparse matrix. I can see the need to maintain the relation between IDs and rows but it should be done by either returning an ID vector or by telling the user in the documentation that the order in the distance matrix is the same as the order in the dataframes that were given to the function. 
- The function should either call or be merged with the trackintel.geogr.distances.calculate_distance_matrix function.

Reply:
edit: The function actually supports sparse calculation